From 09f963a7d09a6414bb20b80be9c72d000c8f0c9a Mon Sep 17 00:00:00 2001
From: Adrien Destugues <pulkomandy@pulkomandy.tk>
Date: Sun, 10 Mar 2019 13:45:11 +0100
Subject: Fix build on x86 32bit

Conflict of macro name with a struct field in one of our headers.

diff --git a/psi/estack.h b/psi/estack.h
index f092eee..75e8d6c 100644
--- a/psi/estack.h
+++ b/psi/estack.h
@@ -18,6 +18,8 @@
 #ifndef estack_INCLUDED
 #  define estack_INCLUDED
 
+#include <signal.h> /* before we #define esp, which prevents it from building */
+
 #include "iestack.h"
 #include "icstate.h"		/* for access to exec_stack */
 
-- 
2.51.0


From 6f6d82bec3c385d333e802ac23385ac207d5c1f5 Mon Sep 17 00:00:00 2001
From: Adrien Destugues <pulkomandy@pulkomandy.tk>
Date: Wed, 13 Mar 2019 21:14:28 +0100
Subject: fix building shared libs


diff --git a/configure.ac b/configure.ac
index ddd0e76..3788fdf 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3232,7 +3232,7 @@ case $host in
       GPDL_SONAME_MAJOR="lib\$(GPDL${libname3}"
       GPDL_SONAME_MAJOR_MINOR="cyg\$(GPDL${libname2}"
     ;;
-    *bsd*)
+    *bsd*|*haiku*)
       DYNAMIC_CFLAGS="-fPIC $DYNAMIC_CFLAGS"
       GS_DYNAMIC_LDFLAGS="-shared -Wl,\$(LD_SET_DT_SONAME)\$(LDFLAGS_SO_PREFIX)\$(GS_SONAME_MAJOR)"
       PCL_DYNAMIC_LDFLAGS="-shared -Wl,\$(LD_SET_DT_SONAME)\$(LDFLAGS_SO_PREFIX)\$(PCL_SONAME_MAJOR)"
-- 
2.51.0


From 129fe8e5863698dc2ad3b778aed8288f0ad17c48 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Joachim=20Mairb=C3=B6ck?= <j.mairboeck@gmail.com>
Date: Sun, 31 Oct 2021 22:03:15 +0100
Subject: fix looking for dlopen in libdl only


diff --git a/configure.ac b/configure.ac
index 3788fdf..860c860 100644
--- a/configure.ac
+++ b/configure.ac
@@ -681,8 +681,8 @@ else
   GCFLAGS="$GCFLAGS -DHAVE_INTTYPES_H=0"
 fi
 
-AC_CHECK_LIB([dl], [dlopen],
-             [AC_CHECK_HEADER([dlfcn.h], [GCFLAGS="$GCFLAGS -DHAVE_LIBDL=1";LIBS="-ldl $LIBS"])]
+AC_SEARCH_LIBS([dlopen], [dl],
+             [AC_CHECK_HEADER([dlfcn.h], [GCFLAGS="$GCFLAGS -DHAVE_LIBDL=1"])]
               )
 
 large_color_index=1
-- 
2.51.0


From 42018ac2176d0d202ead20bf6378cbfdee76acef Mon Sep 17 00:00:00 2001
From: Anarchos <sylvain_kerjean@hotmail.com>
Date: Wed, 2 Mar 2022 21:15:32 +0100
Subject: debug flags for haiku


diff --git a/configure.ac b/configure.ac
index 860c860..3d19bec 100644
--- a/configure.ac
+++ b/configure.ac
@@ -280,7 +280,7 @@ SET_DT_SONAME="-soname="
 CFLAGS_LARGEFILE=""
 
 case $host in
-        *-linux*|*-gnu)
+        *-linux*|*-gnu|*-haiku)
         if test x"$GCC" = x"yes"; then
             CC_OPT_FLAGS_TO_TRY="-O2 -DNDEBUG"
             CC_DBG_FLAGS_TO_TRY="-gdwarf-2 -g3 -O0"
-- 
2.51.0


From 0aa70431ed1dd85d558e6846ffa3637861954053 Mon Sep 17 00:00:00 2001
From: Anarchos <sylvain_kerjean@hotmail.com>
Date: Fri, 4 Mar 2022 05:47:01 +0100
Subject: -g


diff --git a/configure.ac b/configure.ac
index 3d19bec..9ce1d85 100644
--- a/configure.ac
+++ b/configure.ac
@@ -283,7 +283,7 @@ case $host in
         *-linux*|*-gnu|*-haiku)
         if test x"$GCC" = x"yes"; then
             CC_OPT_FLAGS_TO_TRY="-O2 -DNDEBUG"
-            CC_DBG_FLAGS_TO_TRY="-gdwarf-2 -g3 -O0"
+            CC_DBG_FLAGS_TO_TRY="-g"
             CC_VISIBILITY_FLAGS_TO_TRY="-fvisibility=hidden"
         fi
         ;;
-- 
2.51.0


From 407e521eaab831ec24f4a5280ee15265ebc5ab47 Mon Sep 17 00:00:00 2001
From: Anarchos <sylvain_kerjean@hotmail.com>
Date: Thu, 10 Mar 2022 17:11:47 +0100
Subject: debug flags


diff --git a/base/unix-dll.mak b/base/unix-dll.mak
index 7c11d89..c8bd4a0 100644
--- a/base/unix-dll.mak
+++ b/base/unix-dll.mak
@@ -431,7 +431,8 @@ soinstall:
 	$(MAKE) $(SUB_MAKE_OPTION) soinstall-subtarget BUILDDIRPREFIX=$(SODIRPREFIX)
 
 sodebuginstall:
-	$(MAKE) $(SUB_MAKE_OPTION) soinstall-subtarget GENOPT='-DDEBUG' BUILDDIRPREFIX=$(SODEBUGDIRPREFIX)
+	GENOPT='-DDEBUG -g'
+	$(MAKE) $(SUB_MAKE_OPTION) soinstall-subtarget GENOPT='$(GENOPT)' BUILDDIRPREFIX=$(SODEBUGDIRPREFIX)
 
 soinstall-subtarget: install-so install-scripts install-data $(INSTALL_SHARED) $(INSTALL_CONTRIB)
 
-- 
2.51.0


From ee629bbc2e8c344d943a487ff1693bf89d144645 Mon Sep 17 00:00:00 2001
From: Anarchos <sylvain_kerjean@hotmail.com>
Date: Thu, 10 Mar 2022 21:44:39 +0100
Subject: revert


diff --git a/base/unix-dll.mak b/base/unix-dll.mak
index c8bd4a0..7c11d89 100644
--- a/base/unix-dll.mak
+++ b/base/unix-dll.mak
@@ -431,8 +431,7 @@ soinstall:
 	$(MAKE) $(SUB_MAKE_OPTION) soinstall-subtarget BUILDDIRPREFIX=$(SODIRPREFIX)
 
 sodebuginstall:
-	GENOPT='-DDEBUG -g'
-	$(MAKE) $(SUB_MAKE_OPTION) soinstall-subtarget GENOPT='$(GENOPT)' BUILDDIRPREFIX=$(SODEBUGDIRPREFIX)
+	$(MAKE) $(SUB_MAKE_OPTION) soinstall-subtarget GENOPT='-DDEBUG' BUILDDIRPREFIX=$(SODEBUGDIRPREFIX)
 
 soinstall-subtarget: install-so install-scripts install-data $(INSTALL_SHARED) $(INSTALL_CONTRIB)
 
-- 
2.51.0

